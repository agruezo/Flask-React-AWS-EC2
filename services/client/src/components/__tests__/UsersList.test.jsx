/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, cleanup } from "@testing-library/react";

import UsersList from "../UsersList";

afterEach(cleanup);

const users = [
  {
    email: "redninja@dojo.com",
    id: 1,
    username: "redninja",
  },
  {
    email: "blueninja@dojo.com",
    id: 2,
    username: "blueninja",
  },
];

it("renders a username", () => {
  const { getByText } = render(
    <UsersList
      users={users}
      removeUser={() => true}
      isAuthenticated={() => true}
    />
  );
  expect(getByText("redninja")).toHaveClass("username");
  expect(getByText("blueninja")).toHaveClass("username");
});

it("renders", () => {
  const { asFragment } = render(
    <UsersList
      users={users}
      removeUser={() => true}
      isAuthenticated={() => false}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders when authenticated", () => {
  const { asFragment } = render(
    <UsersList
      users={users}
      removeUser={() => true}
      isAuthenticated={() => true}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
