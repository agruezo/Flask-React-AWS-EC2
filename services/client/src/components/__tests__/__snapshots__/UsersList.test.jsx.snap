// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            redninja@dojo.com
          </td>
          <td
            class="username"
          >
            redninja
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            blueninja@dojo.com
          </td>
          <td
            class="username"
          >
            blueninja
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`renders when authenticated 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            redninja@dojo.com
          </td>
          <td
            class="username"
          >
            redninja
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            blueninja@dojo.com
          </td>
          <td
            class="username"
          >
            blueninja
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
